frontend/src/metabase/visualizations/lib/action.unit.spec.ts

```diff
@@ -35,4 +37,184 @@ describe("performAction", () => {
+          payload: {
+            args: [
+              {
+                hash: "",
+                pathname: url,
+                query: {},
+                search: "",
+              },
+            ],
+            method: "push",
+          },
+          type: "@@router/CALL_HISTORY_METHOD",
+        });
+      },
+    );
+
+    it.each(["auto/dashboard/adhoc/123Abc", "auto/dashboard/adhoc/123Abc/"])(
+      `should redirect using router when using a relative URL: "%s"`,
+      url => {
+        MetabaseSettings.set("site-url", sitePath);
+        const action: UrlClickAction = {
+          buttonType: "horizontal",
+          name: "automatic-insights",
+          section: "auto",
+          url: jest.fn(() => url),
+        };
+
+        const extraProps = {
+          dispatch: jest.fn(),
+          onChangeCardAndRun: jest.fn(),
+        };
+
+        expect(performAction(action, extraProps)).toBe(true);
+
+        expect(action.url).toHaveBeenCalledTimes(1);
+
+        expect(extraProps.dispatch).toHaveBeenCalledTimes(2);
+        expect(extraProps.dispatch).toHaveBeenCalledWith({
+          payload: {
+            args: [
+              {
+                hash: "",
+                pathname: "/" + url,
+                query: {},
+                search: "",
+              },
+            ],
+            method: "push",
+          },
+          type: "@@router/CALL_HISTORY_METHOD",
+        });
+      },
+    );
+
+    it("should redirect using router when using an absolute URL containing site URL", () => {
+      MetabaseSettings.set("site-url", sitePath);
+      const action: UrlClickAction = {
+        buttonType: "horizontal",
+        name: "automatic-insights",
+        section: "auto",
+        url: jest.fn(
+          // This ensures that even if the subpath has different cases, it will still work
+          () => `${sitePath.toLowerCase()}/auto/dashboard/adhoc/123Abc`,
+        ),
+      };
+
+      const extraProps = {
+        dispatch: jest.fn(),
+        onChangeCardAndRun: jest.fn(),
+      };
+
+      expect(performAction(action, extraProps)).toBe(true);
+
+      expect(action.url).toHaveBeenCalledTimes(1);
+
+      expect(extraProps.dispatch).toHaveBeenCalledTimes(2);
+      expect(extraProps.dispatch).toHaveBeenCalledWith({
+        payload: {
+          args: [
+            {
+              hash: "",
+              pathname: "/auto/dashboard/adhoc/123Abc",
+              query: {},
+              search: "",
+            },
+          ],
+          method: "push",
+        },
+        type: "@@router/CALL_HISTORY_METHOD",
+      });
+    });
+
+    it("should redirect with invalid URL", () => {
+      MetabaseSettings.set("site-url", sitePath);
+      const action: UrlClickAction = {
+        buttonType: "horizontal",
+        name: "automatic-insights",
+        section: "auto",
+        url: jest.fn(
+          () =>
+            "invalid_protocol://localhost/metabase/auto/dashboard/adhoc/123Abc",
```

Comment:
I was thinking about completely invalid URLs, not valid ones with some unexpected protocol

Reply:
Can you give me some examples?